terraso_backend/apps/auth/providers.py

```diff
@@ -149,3 +149,57 @@ def _build_client_secret(self):
             algorithm=self.JWT_ALGORITHM,
             headers=jwt_header,
         )
+
+
+class MicrosoftProvider:
+    OAUTH_BASE_URL = (
+        f"https://login.microsoft.com/{settings.MICROSOFT_TENANT}/oauth2/v2.0/authorize?"
+    )
+    TOKEN_URI = f"https://login.microsoftonline.com/{settings.MICROSOFT_TENANT}/oauth2/v2.0/token"
```

Comment:
suggestion: these two URLs are almost the same. combine into `MS_BASE_URL` (use a different constant name if you prefer).